forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabMergeRequest.java

```diff
@@ -841,7 +841,16 @@ public Optional<HostUser> closedBy() {
 
     @Override
     public URI filesUrl(Hash hash) {
-        var endpoint = "/" + repository.name() + "/-/merge_requests/" + id() + "/diffs?commit_id=" + hash.hex();
-        return host.getWebUri(endpoint);
+        var versionId = request.get("versions").execute().stream()
+                               .filter(version -> hash.hex().equals(version.get("head_commit_sha").asString()))
+                               .map(version -> String.valueOf(version.get("id").asInt()))
```

Comment:
I don't think we need to convert this to an int to then just create a string again.

Reply:
The `id` field of a `version` is a int type in the json so that `version.get("id").asString()` will throw a exception.